src/test/java/com/epam/task/first/logic/ArrayLogicTest.java

```diff
@@ -0,0 +1,205 @@
+package com.epam.task.first.logic;
+
+import com.epam.task.first.entities.Array;
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class ArrayLogicTest {
+
+    private ArrayLogic arrayLogic =new ArrayLogic();
+
+    @Test
+    public void testFindMaxShouldFindMaxWhenPositiveNumbersApplied(){
+        Array array = new Array(1,3,2);
+
+        int actual = arrayLogic.findMax(array);
+
+        Assert.assertEquals(3,actual);
+    }
+
+    @Test
+    public void testFindMaxShouldFindMaxWhenNegativeNumbersApplied(){
+        Array array = new Array(-1,-3,-2);
+
+        int actual = arrayLogic.findMax(array);
+
+        Assert.assertEquals(-1,actual);
+    }
+
+    @Test
+    public void testFindMaxShouldFindMaxWhenIntegerNumbersApplied(){
+        Array array = new Array(-10,0,50);
+
+        int actual = arrayLogic.findMax(array);
+
+        Assert.assertEquals(50,actual);
+    }
+
+    @Test
+    public void testFindMinShouldFindMinWhenPositiveNumbersApplied(){
+        Array array = new Array(2,5,1);
+
+        int actual = arrayLogic.findMin(array);
+
+        Assert.assertEquals(1,actual);
+    }
+
+    @Test
+    public void testFindMinShouldFindMinWhenNegativeNumbersApplied(){
+        Array array = new Array(-10,-5,-20);
+
+        int actual = arrayLogic.findMin(array);
+
+        Assert.assertEquals(-20,actual);
+    }
+
+    @Test
+    public void testFindMinShouldFindMinWhenIntegerNumbersApplied(){
+        Array array = new Array(-10,0,-50);
+
+        int actual = arrayLogic.findMin(array);
+
+        Assert.assertEquals(-50,actual);
+    }
+
+    @Test
+    public void replaceNonzeroElementsByOnesWithSignShouldReplaceNonzeroElementsByOnesWhenPositiveNumbersApplied(){
+        Array array = new Array(1,2,4,6,20);
+        List<Integer> expected = new Array(1,1,1,1,1).getElements();
+
+        List<Integer> actual = arrayLogic.replaceNonzeroElementsByOnesWithSign(array);
+
+        Assert.assertEquals(expected,actual);
+    }
+
+    @Test
+    public void replaceNonzeroElementsByOnesWithSignShouldReplaceNonzeroElementsByMinusOnesWhenNegativeNumbersApplied(){
+        Array array = new Array(-1,-2,-4,-6,-20);
+        List<Integer> expected = new Array(-1,-1,-1,-1,-1).getElements();
+
+        List<Integer> actual = arrayLogic.replaceNonzeroElementsByOnesWithSign(array);
```

Comment:
if you define method equals in Array class, and from your method you would return Array object, so you can use assertEquals